From a7e2245c75c2fdf444b71cd8f6204248809f98d5 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Fri, 14 Apr 2023 14:20:52 +0200
Subject: [PATCH] frdp-session: Fix crash during resizing of desktop

Check whether display control channel is available before resizing desktop
in configure event handler.

Fixes #41
---
 src/frdp-session.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/frdp-session.c b/src/frdp-session.c
index 93f4c49..5713e6c 100644
--- a/src/frdp-session.c
+++ b/src/frdp-session.c
@@ -263,8 +263,9 @@ frdp_session_configure_event (GtkWidget *widget,
   g_object_get (G_OBJECT (widget), "allow-resize", &allow_resize, NULL);
 
   if (allow_resize) {
-    if (settings->DesktopWidth != gtk_widget_get_allocated_width (scrolled) ||
-        settings->DesktopHeight != gtk_widget_get_allocated_height (scrolled)) {
+    if ((settings->DesktopWidth != gtk_widget_get_allocated_width (scrolled) ||
+         settings->DesktopHeight != gtk_widget_get_allocated_height (scrolled)) &&
+        priv->display_control_channel != NULL) {
       frdp_channel_display_control_resize_display (priv->display_control_channel,
                                                    width,
                                                    height);
-- 
2.47.1

